metaflow/runner/subprocess_manager.py

```diff
@@ -42,6 +42,19 @@ class SubprocessManager(object):
     def __init__(self):
         self.commands: Dict[int, CommandManager] = {}
 
+        try:
+            loop = asyncio.get_running_loop()
+            loop.add_signal_handler(
+                signal.SIGINT,
+                lambda: self._handle_sigint(signum=signal.SIGINT, frame=None),
```

Comment:
I think this would be better with `asyncio.create_task` and a wrapped async signal handler.

By using a synchronous one with `time.sleep` and `subprocess.check_call`, instead of awaiting on `asyncio.sleep` and `asyncio.create_subprocess_exec`, you're blocking other coroutines from running in the meantime.

Reply:
Thanks @filipcacky, would you like to contribute this change?

Reply:
If you want, either you can push on top of my branch / fork..
OR we can merge this and you can raise another.. what say?

Reply:
Hey, sorry, I'm from EU so it's a bit late for me 🙂 I've implemented this in async in my MR. You can copy it over from there.

Reply:
No worries, you can contribute tomorrow as discussed in DM 😄

Reply:
Didn't have permissions to push/create PR to your remote, so I added my changes here https://github.com/filipcacky/metaflow/commit/5dd4aaba43db3f405a1623d05ae25c806c587c9f. Feel free to cherry-pick.